website/src/pages/api/auth/[...nextauth].ts

```diff
@@ -200,6 +200,24 @@ export default function auth(req: NextApiRequest, res: NextApiResponse) {
           select: { name: true, role: true, isNew: true, accounts: true, id: true },
         });
 
+        if (!frontendUser) {
+          // should never reach here
+          throw new Error("User not found");
+        }
+
+        // TODO avoid duplicate logic with the signIn event
+        if (frontendUser.isNew && !frontendUser.name) {
```

Comment:
we already create a username in line 143?

Reply:
Just to ensure everything still works, since we have no test for this...

Reply:
Unfortunately, I still don't see which edge case is being caught here, or why are we checking for the name in line 140? we had the same sign up process for ever, did anything regarding the sign up requirement change?

Reply:
I'm not sure how next auth work internally, but it seem the `jwt` callback is called before the `signIn` event, so in that case the name is not available yet, but the backend requires a name so I have to assign it first. The check `!user.name` on line 140 is to make sure we don't assign the name again

Reply:
ok I see, so the reason is the upsert, can you please comment that in the code, so we know why we did this?

Reply:
Done